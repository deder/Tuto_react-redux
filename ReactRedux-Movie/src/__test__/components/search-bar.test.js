import SearchBar from '../../components/search-bar';
import React from 'react';

describe("<SearchBar />",()=>{
    it("Snapshot", () => {
        const onSearch = jest.fn();
        const toogleShowForm= jest.fn();
        const wrapper = mount(
            <SearchBar onSearch={onSearch} toogleShowForm={toogleShowForm} />
        );
        expect(wrapper).toMatchSnapshot();
    });
    it("onChangeHandler", done => {
        const onSearch = jest.fn();
        const event = {
          preventDefault() {},
          target: { value: 'pokemon' }
        };

        const wrapper = mount(
            <SearchBar onSearch={onSearch} />
        );
        wrapper.find('#search').simulate('change', event);
        setTimeout(()=>{
            try {
                expect(onSearch).toHaveBeenCalled();
                done()
            } catch (error) {
                done.fail(error)
            }
        },1000)
    });
    it("onChangeHandler with double change", done => {
        const onSearch = jest.fn();
        const event = {
          preventDefault() {},
          target: { value: 'pokemon' }
        };
        const event2 = {
            preventDefault() {},
            target: { value: 'terminator' }
          };
        const wrapper = mount(
            <SearchBar onSearch={onSearch} />
        );
        wrapper.find('#search').simulate('change', event);
        wrapper.find('#search').simulate('change', event2);
        setTimeout(()=>{
            try {
              
                expect(onSearch).toHaveBeenCalledWith('pokemon');
                done()
            } catch (error) {
                done.fail(error)
            }
        },1001)
    });

    it("onEnterHandler with key 'Enter'", done => {
        const onSearch = jest.fn();
        const event = {
          preventDefault() {},
          target: { value: 'pokemon' },
          keyCode:13,
          key:"Enter"
        };
        const wrapper = shallow(
            <SearchBar onSearch={onSearch}  />
        );
        wrapper.find('#search').simulate('keyPress', event);
        setTimeout(()=>{
            try {
                expect(onSearch).toHaveBeenCalled();
                done()
            } catch (error) {
                done.fail(error)
            }
        })
    });
    it("onEnterHandler without key 'Enter'", done => {
        const onSearch = jest.fn();
        const event = {
            preventDefault() {},
            target: { value: 'terminator' },
            key:""
          };

        const wrapper = shallow(
            <SearchBar onSearch={onSearch} />
        );
        wrapper.find('#search').simulate('keyPress', event);
        setTimeout(()=>{
            try {
                expect(onSearch).not.toHaveBeenCalled();
                done()
            } catch (error) {
                done.fail(error)
            }
        },500)
    });
})