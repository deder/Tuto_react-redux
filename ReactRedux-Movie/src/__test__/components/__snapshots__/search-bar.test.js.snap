// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> Snapshot 1`] = `
<form
  id="formSearch"
  style={
    Object {
      "backgroundColor": "white",
      "height": "100%",
      "left": 0,
      "position": "absolute",
      "top": 0,
      "width": "100%",
    }
  }
>
  <div>
    <i
      className="material-icons"
      onClick={[Function]}
      style={
        Object {
          "color": "black",
          "cursor": "pointer",
          "paddingLeft": "20px",
          "paddingRight": "20px",
          "position": "absolute",
          "right": 0,
          "zIndex": 5,
        }
      }
    >
      close
    </i>
    <input
      id="search"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Entrer le nom d'un film"
      style={
        Object {
          "color": "grey",
          "paddingLeft": "24px",
          "paddingRight": "24px",
        }
      }
      type="search"
      value=""
    />
  </div>
</form>
`;
